OneCommunityGlobal/HighestGoodNetworkApp
# yixiao_lei_color_discrepancy_on_piechart

# Description
Ensure that each segment of the pie chart is colored according to predefined preferences associated with personal test lists.
Enhance the pie chart by adding interactive tooltips that display the name of the test associated with each segment when the user hovers over it. These tooltips should disappear when the mouse is moved away from the segment.

Fixes # 
<img width="776" alt="Screenshot 2024-03-09 at 22 14 06" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/46af0ac0-657b-45af-a49c-d781db8786b8">
	
## Related PRS (if any):
None
…

## Main changes explained:
- Adjust the Pie Chart Colors to Match Personal Test Lists
- Implement Interactive Tooltips with Mouseover and Mouseout Events

## How to test:
check into current branch on the front end and "development" branch on the backend
do npm install and npm run start:local to run this PR locally
Clear site data/cache
log as admin/owner user
Follow [these instructions](https://www.loom.com/share/8bc672c323b64955aac4bce46c3794aa?sid=b05f5604-87cb-4ca1-80b8-49bd823893ee) to add a user account you want to test to a project and assign at least 10 tasks to that user
The tasks with no hours logged will not be displayed on the Pie Chart. To add hours logged, connect to hgnproddb using Mongo Compass, select the "tasks" collection, search for the task you need by taskName and add some number (not zero) to the "hoursLogged" field.
<img width="1409" alt="Screenshot 2024-03-09 at 22 17 40" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/1d940738-096c-47a8-8511-0fcf35a09bda">

Go to the Home Page -> Reports dropdown -> Reports -> People -> Search to find the user account you assigned tasks to and click on it -> test the functionality by clicking "View all"
Ensure that the colors displayed in the pie chart's legend accurately match the corresponding segments of the pie chart and the tasks on the pie can be shown when mouse over it. 

## Screenshots or videos of changes:
Before:

<img width="715" alt="Screenshot 2024-03-09 at 22 41 47" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/061caee4-8097-4fe0-a3e1-5b85c30645cc">


After

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/af043410-2b1d-4861-9b10-4a4986ee3033







## Note:
None
